snapcraft/plugins/_python/_pip.py

```diff
@@ -129,10 +137,23 @@ def setup(self):
         Check to see if pip has already been installed. If not, fetch pip,
         setuptools, and wheel, and install them so they can be used.
         """
-        # Check to see if we have our own pip, yet. If not, we need to use the
-        # pip on the host (installed via build-packages) to grab our own.
-        if not self.is_setup():
-            logger.info('Fetching pip, setuptools, and wheel...')
+
+        self._ensure_pip_installed()
+        self._ensure_wheel_installed()
+        self._ensure_setuptools_installed()
+
+    def is_setup(self):
```

Comment:
Would this be better: `was_setup` or `is_already_setup`?
is_setup sounds to me like checking if this is of type `setup` or not. Not if `setup` happened.

Reply:
That feels a little overly verbose to me, but I have no strong feelings about it. However, this API was already in place and is used elsewhere, so in the interest of not further inflating this diff I'd rather not make that change here.